File: projet_groupe4/src/main/java/projet_groupe4/dto/response/ClientWithReservationResponse.java
package projet_groupe4.dto.response;

import java.util.List;

import projet_groupe4.model.Client;

public class ClientWithReservationResponse {

    private Integer id;
    private String nom;
    private String prenom;
    private String email;
    private String telephone;
    private List<ReservationResponse> reservations;

    public Integer getId() {
        return id;
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public String getTelephone() {
        return telephone;
    }

    public void setTelephone(String telephone) {
        this.telephone = telephone;
    }

    public void setId(Integer id) {
        this.id = id;
    }

    public String getNom() {
        return nom;
    }

    public void setNom(String nom) {
        this.nom = nom;
    }

    public String getPrenom() {
        return prenom;
    }

    public void setPrenom(String prenom) {
        this.prenom = prenom;
    }

    public List<ReservationResponse> getReservations() {
        return reservations;
    }

    public void setReservations(List<ReservationResponse> reservations) {
        this.reservations = reservations;
    }

    public static ClientWithReservationResponse convert(Client client) {
        ClientWithReservationResponse resp = new ClientWithReservationResponse();

        resp.setId(client.getId());
        resp.setNom(client.getNom());
        resp.setPrenom(client.getPrenom());
        resp.setEmail(client.getMail());
        resp.setTelephone(client.getTelephone());
        resp.setReservations(client.getReservations().stream().map(ReservationResponse::convert).toList());

        return resp;
    }
}

File: projet_groupe4_angular/src/app/dto/client-with-reservation-response.ts
import { Reservation } from './reservation';

export class ClientWithReservationResponse {
  constructor(
    private _id: number,
    private _nom: string,
    private _prenom: string,
    private _email: string,
    private _telephone: string,
    private _reservations: Reservation[]
  ) {}

  public get id(): number {
    return this._id;
  }

  public set id(value: number) {
    this._id = value;
  }

  public get nom(): string {
    return this._nom;
  }

  public set nom(value: string) {
    this._nom = value;
  }

  public get prenom(): string {
    return this._prenom;
  }

  public set prenom(value: string) {
    this._prenom = value;
  }

  public get email(): string {
    return this._email;
  }

  public set email(value: string) {
    this._email = value;
  }

  public get telephone(): string {
    return this._telephone;
  }

  public set telephone(value: string) {
    this._telephone = value;
  }

  public get reservations(): Reservation[] {
    return this._reservations;
  }

  public set reservations(value: Reservation[]) {
    this._reservations = value;
  }

  public toJson(): any {
    return {
      nom: this.nom,
      prenom: this.prenom,
      email: this.email,
      telephone: this.telephone,
      reservations: this.reservations,
    };
  }
}
